packages/flutter/test/cupertino/slider_test.dart

```diff
@@ -556,4 +556,29 @@ void main() {
       isNot(paints..rrect(color: _kSystemFill.color)),
     );
   });
+
+  testWidgets('Thumb color can be overridden', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      CupertinoApp(
+        theme: const CupertinoThemeData(brightness: Brightness.light),
+        home: Center(
+          child: CupertinoSlider(
+            thumbColor: CupertinoColors.activeGreen,
+            onChanged: (double value) { },
+            value: 0,
+          ),
+        ),
+      ),
+    );
+
+    expect(
+      find.byType(CupertinoSlider),
+      paints..rrect(color: _kSystemFill.color),
+    );
+
+    expect(
+      find.byType(CupertinoSlider),
+      isNot(paints..rrect(color: _kSystemFill.darkColor)),
+    );
+  });
```

Comment:
In addition to this, could you also verify that if a `CupertinoSlider` rebuilds with a different `thumbColor`, the thumb color correctly updates?

Reply:
@LongCatIsLooong, when I made the thumbColor update like the trackColor I noticed that my test wasn't actually checking the color of the thumb, instead it was checking the color of the track.

I was thinking in checking in the object if the color had changed, but this wouldn't actually verify that the correct color os painted on the screen. Do you know if there is a way to use the `paints..rrect` to match the thumb so I can check that the correct color was painted?

Reply:
Oh, I figured a way, but I don't know if it is the best way to test this. Since the thumb is the last rect being drawn, I have to call rrect() six times.

```dart
expect(
      find.byType(CupertinoSlider),
      paints
      ..rrect()
      ..rrect()
      ..rrect()
      ..rrect()
      ..rrect()
      ..rrect(color: CupertinoColors.systemPurple.color)
    );
```

Reply:
Yeah it's a bit unfortunate but I think this is fine. The only alternative I know of is a golden test.